repos/system_upgrade/common/actors/addupgradebootentry/libraries/addupgradebootentry.py

```diff
@@ -1,41 +1,153 @@
+import itertools
 import os
 import re
 
 from leapp.exceptions import StopActorExecutionError
 from leapp.libraries.common.config import architecture
 from leapp.libraries.stdlib import api, CalledProcessError, run
-from leapp.models import BootContent, KernelCmdlineArg, TargetKernelCmdlineArgTasks
+from leapp.models import (
+    BootContent,
+    KernelCmdline,
+    KernelCmdlineArg,
+    LiveImagePreparationInfo,
+    LiveModeArtifacts,
+    LiveModeConfig,
+    TargetKernelCmdlineArgTasks
+)
+
+
+def collect_boot_args(livemode_enabled):
+    args = {
+        'enforcing': '0',
+        'rd.plymouth': '0',
+        'plymouth.enable': '0'
+    }
+
+    if os.getenv('LEAPP_DEBUG', '0') == '1':
+        args['debug'] = None
+
+    if os.getenv('LEAPP_DEVEL_INITRAM_NETWORK') in ('network-manager', 'scripts'):
+        args['ip'] = 'dhcp'
+        args['rd.neednet'] = '1'
```

Comment:
Are the reasons for using `get_env` documented anywhere?

Reply:
not sure about documentation, but basic rule we set in the befinning is using leapp library functions first. similar to why people should not use python `subprocess`  but `run` function from leapp stdlib.

Another reasons are:
* os.getenv is not save to be used in any actor - e.g. in initramfs, firstboot, ...). the get_env function is supposed to provide information about any valid LEAPP envar to any actor.
* even when some actors could use os.getenv without seeing a difference, it's inconsistent when another actors must really use the get_env function instead
* no need to play with real defined envars in unit-tests, as this is handled by CurrentActorMocked - where if the "envar" is set, it's expected that code in actor will work still corectly during unit-tests execution - which is not true when using os.getenv again. so again, here would be another inconsistency.

Reply:
and most likely we should update the docstring of the function as well

Reply:
btw, to be clear, this is not blocker for the merge as there are already actors using the `os.getenv` present in the repo, so it can be polished in a separate PR that will deal with all of them. but I wrote it to let you know about that

Reply:
@MichalHe  thanks!